instrumentation/play/play-2.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/Play24Singletons.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.play.v2_4;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.tracer.ServerSpan;
+import play.api.mvc.Request;
+import scala.Option;
+
+public final class Play24Singletons {
+
+  private static final String SPAN_NAME = "play.request";
+  private static final Instrumenter<Void, Void> INSTRUMENTER =
+      Instrumenter.<Void, Void>builder(
+              GlobalOpenTelemetry.get(), "io.opentelemetry.play-2.4", s -> SPAN_NAME)
+          .newInstrumenter();
+
+  public static Instrumenter<Void, Void> instrumenter() {
+    return INSTRUMENTER;
+  }
+
+  public static void updateSpanNames(Context context, Request<?> request) {
+    String route = getRoute(request);
+    if (route == null) {
+      return;
+    }
+
+    Span.fromContext(context).updateName(route);
+    // set the span name on the upstream akka/netty span
+    Span serverSpan = ServerSpan.fromContextOrNull(context);
+    if (serverSpan != null) {
+      serverSpan.updateName(route);
```

Comment:
WDYT about using `ServerSpanNaming` here? Netty 4.x instrumentations actually initialize it, and I think ratpack instrumentation uses it to update server span names -- I think it should probably work here too.

Reply:
When I realized that nothing tests this I decided to modify it as little as possible.